crates/eframe/src/native/file_storage.rs

```diff
@@ -10,15 +10,60 @@ use std::{
 /// [`egui::ViewportBuilder::app_id`] of [`crate::NativeOptions::viewport`]
 /// or the title argument to [`crate::run_native`].
 ///
-/// On native the path is picked using [`directories::ProjectDirs::data_dir`](https://docs.rs/directories/5.0.1/directories/struct.ProjectDirs.html#method.data_dir) which is:
+/// On native, if the `directories` feature is enabled, the path is picked using
+/// [`directories::ProjectDirs::data_dir`](https://docs.rs/directories/5.0.1/directories/struct.ProjectDirs.html#method.data_dir)
+/// which is:
 /// * Linux:   `/home/UserName/.local/share/APP_ID`
 /// * macOS:   `/Users/UserName/Library/Application Support/APP_ID`
-/// * Windows: `C:\Users\UserName\AppData\Roaming\APP_ID`
+/// * Windows: `C:\Users\UserName\AppData\Roaming\APP_ID\data`
+///
+/// If the `directories` feature is not enabled, it uses a naive approximation that returns the
+/// same result for the most common systems.
 pub fn storage_dir(app_id: &str) -> Option<PathBuf> {
+    #[cfg(feature = "directories")]
+    {
+        directories_storage_dir(app_id)
+    }
+    #[cfg(not(feature = "directories"))]
+    {
+        naive_storage_dir(app_id)
+    }
+}
+
+#[cfg(feature = "directories")]
+#[inline]
+fn directories_storage_dir(app_id: &str) -> Option<PathBuf> {
     directories::ProjectDirs::from("", "", app_id)
         .map(|proj_dirs| proj_dirs.data_dir().to_path_buf())
 }
 
+#[allow(dead_code)]
+#[inline]
+fn naive_storage_dir(app_id: &str) -> Option<PathBuf> {
+    use egui::os::OperatingSystem as OS;
+    use std::env::var_os;
+    match OS::from_target_os() {
+        OS::Nix => var_os("XDG_DATA_HOME")
+            .map(PathBuf::from)
+            .or_else(|| var_os("HOME").map(|s| PathBuf::from(s).join(".local").join("share")))
+            .map(|p| {
+                p.join(
+                    app_id
+                        .to_lowercase()
+                        .replace(|c: char| c.is_ascii_whitespace(), ""),
```

Comment:
This doesn't seems enough to sanitize the path 😬 

https://stackoverflow.com/questions/1976007/what-characters-are-forbidden-in-windows-and-linux-directory-names

I suggest we replace all characters not in `[a-zA-Z0-9_-]`

Reply:
This is actually just replicating what the directories crate does.

The Linux impl changes the text to lowercase and removes spaces:

<https://github.com/dirs-dev/directories-rs/blob/da65f064a8bb18d790600433fadeed17a54c793a/src/lin.rs#L92-L110>

The macOS impl changes the spaces to dashes:

<https://github.com/dirs-dev/directories-rs/blob/da65f064a8bb18d790600433fadeed17a54c793a/src/mac.rs#L91-L98>

And the Windows impl doesn't change anything:

<https://github.com/dirs-dev/directories-rs/blob/da65f064a8bb18d790600433fadeed17a54c793a/src/win.rs#L98-L100>

I suppose we could be more strict, but for now I'm just trying to remove the dependency with minimal changes to the functionality.